src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -339,13 +347,15 @@ protected override async Task<StorageContent> OnLoadAsync(Uri resourceUri, Cance
             }
         }
 
-        protected override async Task OnDeleteAsync(Uri resourceUri, CancellationToken cancellationToken)
+        protected override async Task OnDeleteAsync(Uri resourceUri, DeleteRequestOptions deleteRequestOptions, CancellationToken cancellationToken)
         {
             string name = GetName(resourceUri);
 
+            var accessCondition = (deleteRequestOptions as DeleteRequestOptionsWithAccessCondition)?.AccessCondition;
```

Comment:
>i [](start = 26, length = 1)

default to empty condition?

Reply:
Is there a difference between null and `GenerateEmptyCondition`?
I wanted to avoid affecting the existing code path as much as possible.

Reply:
Check in ILSpy. I believe not. Just saying this is inconsistent with the `??` in your changeset above.

Reply:
I'm going to use `null` to be consistent with the existing behavior when an ETag is not supplied. The `ClockBlockBlob` docs say "If null, no condition is used.".